enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,318 @@
+
+Also the DHCP options will be configured to deliver the address to the VMs
+
+```text
+    ┌──────────────────────┐   ┌────────────────────┐
+    │logical switch node1  │   │logical switch node2│
+    └──────────────────────┘   └────────────────────┘
+┌──────────────────────┐  │     │   ┌──────────────────────┐      
+│ lsp ns-virt-launcher1│──┘     └───│ lsp ns-virt-launcher2│     
+│ dhcpv4_options: 1234 │            │ dhcpv4_options: 1234 │
+│ address:             │            │ address:             │    
+│  0a:58:0a:f4:00:01   │            │  0a:58:0a:f4:00:01   │
+│  10.244.0.8          │            │  10.244.0.8          │
+└──────────────────────┘            └──────────────────────┘
+
+┌─────────────────────────────────┐
+│ dhcp-options 1234               │
+│   lease_time: 3500              │
+│   router: 169.254.1.1           │
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+The CNI will not set an address at virt-launcher pod netns, that address is
+assigned to the VM with the DHCP options from the LSP, this allows to use 
+kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The point to point routing feature allows an address to be running at a node different from
+the one "owning" the subnet the address is coming from. This will happen
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what IP pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement is implemented as part of ovn-k pod's CNI and controllers so
+the feature detection mechanism have to be implemented as part of kubevirt
+virt-launcher pods.
+
+To skip CNI IP configuration and assign assign DHCP options to LSP the following
+label and annotation are check to ensure that only kubevirt pods with explicit configuration
+are affected:
+- `ovn.org/skip-ip-configuration-on-cni`
+- `kubevirt.io/vm`
+
+As part of this enhancement ovn-k should configure a point to point routing after 
+live migration to redirect it to the correct node and port, this feature is only
+activated if the pod is annotated with `kubevirt.io/allow-pod-bridge-network-live-migration`.
+
+During live migration there is source and target virt-launcher pod's with different names, the ovn-k pod's controller watch the following
```

Comment:
wait so if the source and target pods have different names, how does OVNK know source pod the target is migrating from?

Reply:
We check `CreationTimestamp` the feature only care about newest virt-launcher, the old ones is considered leftover and ignored it, I will add that to the enhancement.

https://github.com/ovn-org/ovn-kubernetes/pull/3478/files#diff-eb30366251cb823a9618bac3c224e92aa52cbedf5fa9c0f5a2a4360902540a42R59

Reply:
I think what I am missing is how ovnk knows the linkage between old and new, since they have different names. A timetamp only tells you what is old and what is new, but not which 2 pods are related. @maiqueb suggested:

"I think it'll see two pods (launcher-vmA-<xyz> and launcher-vmA-<zyx>) having [kubevirt.io/vm](http://kubevirt.io/vm) = vmA (same value). Thus it will know it's a migration."

That makes sense if we are using the value of that label to identify the old/new. Can you update the documents to explain that? Maybe show an example?

Reply:
The example would be helpful indeed.

Reply:
> I think what I am missing is how ovnk knows the linkage between old and new, since they have different names. A timetamp only tells you what is old and what is new, but not which 2 pods are related. @maiqueb suggested:
> 
> 
> 
> "I think it'll see two pods (launcher-vmA-<xyz> and launcher-vmA-<zyx>) having [kubevirt.io/vm](http://kubevirt.io/vm) = vmA (same value). Thus it will know it's a migration."
> 
> 
> 
> That makes sense if we are using the value of that label to identify the old/new. Can you update the documents to explain that? Maybe show an example?

We need to you use both things, `kubevirt.io/vm` to search for realted virt-launcher pod and `CreationTimestamp` to sort them out and discover the target virt-launcher pod.